portfolio/src/main/webapp/script.js

```diff
@@ -58,11 +58,16 @@ function handleResponse(response) {
 function addTextToDom(text) {
     console.log("Adding text to DOM: "+text);
     const table = document.getElementById("comment-section");
-    const parsedText = JSON.parse(text);
+    const parsedText = JSON.parse(text); // this is the object with properties numComments and comments
+    parsedText.comments = JSON.parse(parsedText.comments);
+    sessionStorage["size"] = parsedText.comments.length;
 
     var i;
-    for (i = 0; i < parsedText.length; i++) {
-        const comment = parsedText[i];
+    const currPage = sessionStorage["current-page"];
+    const startComment = (currPage - 1) * sessionStorage["num-comments"];
```

Comment:
This calculation should be done in getMessages and passed to the server to only return the comments we care about. If there are 1million comments in the server, we don't want the browser trying to hold all of them in local memory.

Reply:
is it ok to do this with a query string?